test-five/deploy.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Script used for local tests.
+# Note that below env variables need to be informed in order to make the deployment work.
+
+cd frontend
```

Comment:
can you explain why you would take this shell approach instead of npm scripts?

Reply:
I originally created this script solely for local infrastructure testing. However, I later decided to incorporate it into the project as a helpful tool. It now makes complete sense to automate the process and include it in the package.json scripts section within the infrastructure folder for use in different environments, such as e2e, testing, and production.

```json
"build:frontend": "cd ../frontend && npm run build",
"deploy:e2e": "npm run build:frontend && cdk deploy TestFiveStack --parameters Environment=e2e --parameters SnsRecipient=moshtix-infra-e2e@gmail.com --require-approval never --progress=events",
"deploy:testing": "npm run build:frontend && cdk deploy TestFiveStack --parameters Environment=testing --parameters SnsRecipient=moshtix-infra-test@gmail.com --require-approval never --progress=events",
"deploy:production": "npm run build:frontend && cd ../infrastructure && cdk deploy TestFiveStack --parameters Environment=production --parameters SnsRecipient=moshtix-infra-prod@gmail.com --require-approval never --progress=events"
```

Here's an example of how you can create the testing infrastructure:

```zsh
DOMAIN_NAME="example.com" CDK_DEFAULT_ACCOUNT=123456789 CDK_DEFAULT_REGION="ap-southeast-2" npm run deploy:testing
```

This script can also be seamlessly integrated into your CI/CD workflow, as the required environment variables are already included. Additionally, you can leverage AWS SSM to retrieve parameters if needed.rameters.